src/UiPath.Workflow.Runtime/Expressions/CompiledExpressionInvoker.cs

```diff
@@ -244,6 +258,8 @@ private void ProcessLocationReferences()
                 CreateRequiredArguments(requiredLocationNames);
             }
         }
+
+        _locationsInitialized = true;
```

Comment:
_locationsInitialized = true; should be set at the beginning of the method to ensure the previous behavior